packages/manager/modules/billing/src/components/commitment/commitment.controller.js

```diff
@@ -0,0 +1,156 @@
+    $window,
+    atInternet,
+    BillingService,
+    BillingCommitmentService,
+    ovhPaymentMethod,
+    RedirectionService,
+  ) {
+    this.$http = $http;
+    this.$q = $q;
+    this.$translate = $translate;
+    this.$window = $window;
+    this.atInternet = atInternet;
+    this.BillingService = BillingService;
+    this.BillingCommitmentService = BillingCommitmentService;
+    this.ovhPaymentMethod = ovhPaymentMethod;
+    this.RedirectionService = RedirectionService;
+  }
+
+  $onInit() {
+    this.isLoadingService = true;
+    this.paymentMethod = null;
+    this.model = {
+      duration: null,
+      engagement: null,
+      paymentMethod: null,
+    };
+    return this.$q
+      .all({
+        service: this.BillingService.getService(this.serviceId),
+        engagement: this.BillingService.getEngagement(this.serviceId),
+        options: this.BillingService.getOptions(this.serviceId),
+      })
+      .then(({ service, engagement, options }) => {
+        this.service = service;
+        this.service.addOptions(options);
+        this.engagement = engagement;
+        return this.BillingCommitmentService.getCatalogPrice(
+          this.service,
+          this.me,
+        );
+      })
+      .then((defaultPrice) => {
+        this.defaultPrice = defaultPrice;
+        return this.getAvailableEngagements();
+      })
+      .catch((error) => {
+        this.error = error;
+      })
+      .finally(() => {
+        this.isLoadingService = false;
+      });
+  }
+
+  generatePaymentMethodUrl() {
+    const callbackUrl = `${window.location.href}?duration=${this.model.duration.duration}`;
+    this.addPaymentMethodUrl = `${this.RedirectionService.getURL(
+      'addPaymentMethod',
+    )}?callbackUrl=${encodeURIComponent(callbackUrl)}`;
+  }
+
+  getAvailableEngagements() {
+    return this.BillingCommitmentService.getServiceAvailableEngagements(
+      this.service,
+    )
+      .then((availableEngagements) => {
+        this.availableEngagements = groupBy(
+          availableEngagements,
+          'durationInMonths',
+        );
+        this.availableDurations = map(
+          this.availableEngagements,
+          (commitment, duration) =>
+            new CommitmentDuration(duration, commitment, this.defaultPrice),
+        );
+        this.model.duration = this.availableDurations.find(
+          (duration) => duration.duration === this.duration,
+        );
+      })
+      .catch((error) => {
+        this.error = error;
+      });
+  }
+
+  onDurationChange(duration) {
+    const commitments = this.availableEngagements[duration.duration];
+
+    if (this.engagement) {
+      [this.model.engagement] = commitments;
+    }
+
+    const upfront = commitments.find((commitment) => commitment.isUpfront());
+    const periodic = commitments.find((commitment) => commitment.isPeriodic());
+
+    if (upfront && periodic) {
+      this.discount = Math.floor(
+        (periodic.totalPrice.value / upfront.totalPrice.value - 1) * 100,
```

Comment:
if you want to round down a negative number use Math.ceil ;)
```
x = (41 / 51 - 1) * 100 = -25.45
Math.floor(x) = -26
```

Reply:
@frenautvh thanks for the tip but if the number is negative then we have an API issue :)